marlowe-playground-client/src/HaskellEditor/View.purs

```diff
@@ -146,19 +146,32 @@ panelContents state StaticAnalysisView =
   section
     [ classes [ ClassName "panel-sub-header", aHorizontal, Classes.panelContents ]
     ]
-    [ analysisResultPane SetIntegerTemplateParam state
-    , analyzeButton loadingWarningAnalysis analysisEnabled "Analyse for warnings" AnalyseContract
-    , analyzeButton loadingReachability analysisEnabled "Analyse reachability" AnalyseReachabilityContract
-    , analyzeButton loadingCloseAnalysis analysisEnabled "Analyse for refunds on Close" AnalyseContractForCloseRefund
-    ]
+    ( [ analysisResultPane SetIntegerTemplateParam state
+      , analyzeButton loadingWarningAnalysis analysisEnabled "Analyse for warnings" AnalyseContract
+      , analyzeButton loadingReachability analysisEnabled "Analyse reachability" AnalyseReachabilityContract
+      , analyzeButton loadingCloseAnalysis analysisEnabled "Analyse for refunds on Close" AnalyseContractForCloseRefund
+      , clearButton clearEnabled "Clear" ClearAnalysisResults
+      ]
+        <> (if isCompiled then [] else [ div [ classes [ ClassName "choice-error" ] ] [ text "Haskell code needs to be compiled in order to run static analysis" ] ])
+    )
   where
   loadingWarningAnalysis = state ^. _analysisState <<< _analysisExecutionState <<< to isStaticLoading
 
   loadingReachability = state ^. _analysisState <<< _analysisExecutionState <<< to isReachabilityLoading
 
   loadingCloseAnalysis = state ^. _analysisState <<< _analysisExecutionState <<< to isCloseAnalysisLoading
 
-  analysisEnabled = not loadingWarningAnalysis && not loadingReachability && not loadingCloseAnalysis && not (isCompiling state)
+  noneAskedAnalysis = state ^. _analysisState <<< _analysisExecutionState <<< to isNoneAsked
+
+  anyAnalysisLoading = loadingWarningAnalysis || loadingReachability || loadingCloseAnalysis
+
+  analysisEnabled = not anyAnalysisLoading && isCompiled
+
+  clearEnabled = not (anyAnalysisLoading || noneAskedAnalysis)
+
+  isCompiled = case view _compilationResult state of
+    Success (Right _) -> true
+    _ -> false
```

Comment:
no big deal and I think we are all still slightly confused about `has` but it feels like it would look nice here
`has (_compilationResult <<< _Success <<< _Right) state`